src/xds/client.rs

```diff
@@ -266,6 +265,26 @@ impl AdsClient {
                 tokio::time::sleep(backoff).await;
                 backoff
             }
+            Err(e @ Error::GrpcStatus(_)) => {
+                let err_detail = e.to_string();
+                let status: tonic::Status = e.try_into().unwrap();
```

Comment:
unwrap is not needed here, we can do:
```
            Err(ref e @ Error::GrpcStatus(ref status)) => {
                let err_detail = e.to_string();
```

and remove the TryInto impl above

Reply:
good , i didn't know ref usage